src/AzureExtension/DataManager/AzureDataManagerCache.cs

```diff
@@ -0,0 +1,316 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System.Dynamic;
+using DevHomeAzureExtension.Client;
+using DevHomeAzureExtension.DataManager;
+using DevHomeAzureExtension.DataModel;
+using DevHomeAzureExtension.DeveloperId;
+using Microsoft.TeamFoundation.Core.WebApi;
+using Microsoft.TeamFoundation.SourceControl.WebApi;
+using Microsoft.VisualStudio.Services.Account;
+using Microsoft.VisualStudio.Services.Account.Client;
+using Microsoft.VisualStudio.Services.Common;
+using Microsoft.VisualStudio.Services.WebApi;
+using Serilog;
+
+namespace DevHomeAzureExtension;
+
+public partial class AzureDataManager
+{
+    private static readonly Uri _vSAccountUri = new(@"https://app.vssps.visualstudio.com/");
+
+    private static readonly int _pullRequestProjectLimit = 50;
+
+    private static readonly int _pullRequestRepositoryLimit = 25;
+
+    public async Task UpdateDataForAccountsAsync(RequestOptions? options = null, Guid? requestor = null)
+    {
+        // A parameterless call will always update the data, effectively a 'force update'.
+        await UpdateDataForAccountsAsync(TimeSpan.MinValue, options, requestor);
+    }
+
+#pragma warning disable CS1998 // Async method lacks 'await' operators and will run synchronously
+    public async Task UpdateDataForAccountsAsync(TimeSpan olderThan, RequestOptions? options = null, Guid? requestor = null)
+#pragma warning restore CS1998 // Async method lacks 'await' operators and will run synchronously
+    {
```

Comment:
This is a really long method. I'm going to leave a few suggestions to shorten it.

Reply:
Good suggestions, all were taken.